### PR TITLE
ervice Fabric Managed Clusters - api version 2022-08-01-preview

### PR DESCRIPTION
**Pull request by a-santamaria**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [x] PowerShell
      - [x] CLI
      - [x] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following appy to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabric/preview/2022-08-01-preview/nodetype.json [hunk 1]
```diff
@@ -0,0 +1,1564 @@
...
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "force": {
+          "description": "Force the action to go through.",
+          "type": "boolean"
+        },
+        "udWiseUpdate": {
```
Comment (zizw123) [inline specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabric/preview/2022-08-01-preview/nodetype.json:716]: >udWiseUpdate [](http://example.com/codeflow?start=9&length=12)

Consider if you can use enum over boolean, see [https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html)

"Replace boolean/switch properties with better enum"

**Comment by weidongxu-microsoft** [comment]
what's the change since 2022-06-01?

**Comment by a-santamaria** [comment]
> what's the change since 2022-06-01?
@weidongxu-microsoft below are the changes we are introducing on this api:

New properties:
	Node Type properties
		○ VmImageResourceId
		○ FrontendConfigurations.ApplicationGatewayBackendAddressPoolId

Node Type Actions:
	○ NodeTypeActionParameters.UDWiseUpdate

File: specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabric/preview/2022-08-01-preview/nodetype.json (hunk 1 above)
Comment (a-santamaria) [reply specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabric/preview/2022-08-01-preview/nodetype.json:716]: Ok thanks for the suggestion, We will remove this parameter from this api and introduce it next month on a new api version as an enum

**Comment by weidongxu-microsoft** [comment]
> > what's the change since 2022-06-01?
> > @weidongxu-microsoft below are the changes we are introducing on this api:
> 
> New properties: Node Type properties ○ VmImageResourceId ○ FrontendConfigurations.ApplicationGatewayBackendAddressPoolId
> 
> Node Type Actions: ○ NodeTypeActionParameters.UDWiseUpdate

Next time, please have one commit that contains these essential changes along. Don't mix it with api-version update or anything else. (I can find it in your 2nd commit, but it is just hard to review with other changes to dozens of examples.

File: specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabric/preview/2022-08-01-preview/nodetype.json [hunk 2]
```diff
@@ -899,6 +910,10 @@
         "evictionPolicy": {
           "$ref": "#/definitions/EvictionPolicyType",
           "description": "Specifies the eviction policy for virtual machines in a SPOT node type. Default is Delete."
+        },
+        "vmImageResourceId": {
+          "type": "string",
```
Comment (weidongxu-microsoft) [inline specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabric/preview/2022-08-01-preview/nodetype.json:915]: you might want to add http://azure.github.io/autorest/extensions/#x-ms-arm-id-details

same for other one that looks like ARM resource id.

**Review by weidongxu-microsoft — APPROVED** [review]

**Comment by a-santamaria** [comment]
> > > what's the change since 2022-06-01?
> > > @weidongxu-microsoft below are the changes we are introducing on this api:
> > 
> > 
> > New properties: Node Type properties ○ VmImageResourceId ○ FrontendConfigurations.ApplicationGatewayBackendAddressPoolId
> > Node Type Actions: ○ NodeTypeActionParameters.UDWiseUpdate
> 
> Next time, please have one commit that contains these essential changes along. Don't mix it with api-version update or anything else. (I can find it in your 2nd commit, but it is just hard to review with other changes to dozens of examples.

sure, will do that in the future

File: specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabric/preview/2022-08-01-preview/nodetype.json (hunk 2 above)
Comment (a-santamaria) [reply specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabric/preview/2022-08-01-preview/nodetype.json:911]: Thanks, added. Will it be considered a breaking change if we add that for parameters introduced in older apis?

Comment (weidongxu-microsoft) [reply specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabric/preview/2022-08-01-preview/nodetype.json:919]: I am not so sure about existing properties.

Currently there are valuable for track2 .NET SDK. Loop in @ArthurMa1978 for .NET

Comment (ArthurMa1978) [reply specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabric/preview/2022-08-01-preview/nodetype.json:919]: @a-santamaria can you provide the list of `arm-id` property / parameter?

**Comment by a-santamaria** [comment]
@weidongxu-microsoft the api is now available, please merge the pr.

**Pull request merged**